src/Playback/__tests__/Playback.test.jsx

```diff
@@ -0,0 +1,133 @@
+import "@testing-library/jest-dom";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { describe, it, expect, beforeEach, vi } from "vitest";
+
+import ChannelPlayer from "@/Playback/components/ChannelPlayer";
+import useChannelPlayback from "@/Playback/hooks/useChannelPlayback";
+import useUpdateSetting from "@/User/hooks/useUpdateSetting";
+import { useUserStore } from "@/User/stores/useUserStore";
+
+let mockIsChannelChanged = false;
+let mockIsPlaying = false;
+
+vi.mock("@/Channel/stores/useChannelStore", () => ({
+  useChannelStore: (selector) =>
+    selector({
+      isChannelChanged: mockIsChannelChanged,
+    }),
+}));
+
+vi.mock("@/User/stores/useUserStore", () => ({
+  useUserStore: vi.fn(),
+}));
+
+vi.mock("@/Playback/hooks/useChannelPlayback", () => ({
+  default: vi.fn(),
+}));
+
+vi.mock("@/User/hooks/useUpdateSetting", () => ({
+  default: vi.fn(),
+}));
+
+vi.mock("@/User/constants/settingOptions", async (importOriginal) => {
+  const actual = await importOriginal();
+  return {
+    ...actual,
+    SETTING_TITLES: {
+      ...actual.SETTING_TITLES,
+      AD_DETECT: "광고 감지",
+      RETURN_CHANNEL: "기존 채널로 이동",
+    },
+  };
+});
+
+describe("ChannelPlayer", () => {
+  const mockUpdateSetting = vi.fn();
+  const mockHandlePlayPause = vi.fn();
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    useUserStore.mockReturnValue({
+      settings: {
+        RETURN_CHANNEL: true,
+        AD_DETECT: true,
+      },
+    });
+
+    useChannelPlayback.mockReturnValue({
+      selectedChannel: {
+        name: "테스트 채널",
+        logoUrl: "https://example.com/test.jpg",
+      },
+      isPlaying: mockIsPlaying,
+      handlePlayPause: mockHandlePlayPause,
+    });
+
+    useUpdateSetting.mockReturnValue(mockUpdateSetting);
+  });
+
+  it("초기 렌더링 시 채널 이름과 썸네일을 표시한다", () => {
+    mockIsChannelChanged = true;
+    render(<ChannelPlayer />);
+    expect(screen.getByText("테스트 채널")).toBeInTheDocument();
+    expect(screen.getByAltText("테스트 채널 썸네일")).toBeInTheDocument();
+  });
+
+  it("isChannelChanged가 false이면 '광고 감지'를 표시한다", () => {
+    mockIsChannelChanged = false;
+    render(<ChannelPlayer />);
+    expect(screen.getByText("광고 감지")).toBeInTheDocument();
+  });
+
+  it("isChannelChanged가 true이면 '기존 채널로 이동'을 표시한다", () => {
+    mockIsChannelChanged = true;
+    render(<ChannelPlayer />);
+    expect(screen.getByText("기존 채널로 이동")).toBeInTheDocument();
+  });
+
+  it("토글 버튼 클릭 시 updateSetting을 호출한다", () => {
+    mockIsChannelChanged = true;
+    render(<ChannelPlayer />);
+    const toggleButton = screen.getByTestId("toggle-button");
+    fireEvent.click(toggleButton);
+    expect(mockUpdateSetting).toHaveBeenCalled();
+  });
```

Comment:
```suggestion
  it("토글 버튼 클릭 시 updateSetting을 호출한다", () => {
    mockIsChannelChanged = true;

    render(<ChannelPlayer />);

    const toggleButton = screen.getByTestId("toggle-button");
    fireEvent.click(toggleButton);

    expect(mockUpdateSetting).toHaveBeenCalled();
  });
```

테스트 코드들을 하나 하나 자세하게 잘 작성하신 것 같아요! 다만 전체적인 테스트 코드에 개행이 들어가 있으면 더 보기 좋을 것 같습니다!

Reply:
피드백 감사합니다!
말씀해주신 대로 개행을 추가해서 가독성도 신경 써보겠습니다!